airbyte-commons/src/main/java/io/airbyte/commons/json/JsonPaths.java

```diff
@@ -0,0 +1,278 @@
+ * This class wraps that implementation to make it easier to leverage this tool internally.
+ *
+ * GOTCHA: Keep in mind with JSONPath, depending on the query, 0, 1, or N values may be returned.
+ * The pattern for handling return values is very much like writing SQL queries. When using it, you
+ * must consider what the number of return values for your query might be. e.g. for this object: {
+ * "alpha": [1, 2, 3] }, this JSONPath "$.alpha[*]", would return: [1, 2, 3], but this one
+ * "$.alpha[0]" would return: [1]. The Java interface we place over this query system defaults to
+ * returning a list for query results. In addition, we provide helper functions that will just
+ * return a single value (see: {@link JsonPaths#getSingleValue(JsonNode, String)}). These should
+ * only be used if it is not possible for a query to return more than one value.
+ */
+public class JsonPaths {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JsonPaths.class);
+
+  // set default configurations at start up to match our JSON setup.
+  static {
+    Configuration.setDefaults(new Configuration.Defaults() {
+
+      // allows us to pass in Jackson JsonNode
+      private static final JsonProvider jsonProvider = new JacksonJsonNodeJsonProvider();
+      private static final MappingProvider mappingProvider = new JacksonMappingProvider();
+
+      @Override
+      public JsonProvider jsonProvider() {
+        return jsonProvider;
+      }
+
+      @Override
+      public MappingProvider mappingProvider() {
+        return mappingProvider;
+      }
+
+      @Override
+      public Set<Option> options() {
+        /*
+         * All JsonPath queries will return a list of values. This makes parsing the outputs much easier. In
+         * cases where it is not a list, helpers in this class can assert that. See
+         * https://github.com/json-path/JsonPath in the JsonPath documentation.
+         */
+        return EnumSet.of(Option.ALWAYS_RETURN_LIST);
+      }
+
+    });
+  }
+
+  /*
+   * This version of the JsonPath Configuration object allows queries to return to the path of values
+   * instead of the values that were found.
+   */
+  private static final Configuration GET_PATHS_CONFIGURATION = Configuration.defaultConfiguration().addOptions(Option.AS_PATH_LIST);
+
+  /**
+   * Attempt to validate if a string is a valid JSONPath string. This assertion does NOT handle all
+   * cases, but at least a common on. We can add to it as we detect others.
+   *
+   * @param jsonPath - path to validate
+   */
+  public static void assertIsJsonPath(final String jsonPath) {
+    Preconditions.checkArgument(jsonPath.startsWith("$"));
+  }
+
+  /**
+   * Attempt to detect if a JSONPath query could return more than 1 value. This assertion does NOT
+   * handle all cases, but at least a common on. We can add to it as we detect others.
+   *
+   * @param jsonPath - path to validate
+   */
+  public static void assertIsSingleReturnQuery(final String jsonPath) {
```

Comment:
I haven't been through the entire review  but don't we want to do that on the arrays?